dora/core/server/worker/src/main/java/alluxio/worker/netty/PacketReadTaskStateMachine.java

```diff
@@ -65,6 +65,11 @@ public class PacketReadTaskStateMachine<T extends ReadRequestContext<?>> {
   private final BlockingQueue<SendDataEvent> mFlowControlQueue =
       new ArrayBlockingQueue<>((int) MAX_PACKETS_IN_FLIGHT);
 
+  private final BlockingQueue<String> mWaitForTerminatedSignalQueue =
```

Comment:
i was thinking about a `SynchronousQueue`. https://stackoverflow.com/a/8591681
I guess a `ArrayBlockingQueue(1)` is better because the putter doesn't need to block.

Alternatively, what about a `ReentrantLock.Condition`?

Reply:
I am just thinking we can put some useful information (e.g. timestamp, more useful message about the packet sending event) the queue in the future. Maybe we can reuse the information to adjust the `mFlowControlQueue` size dynamically to achieve higher performance. 

As we have already created a `BlockingQueue` (`mFlowControlQueue`) in this class, to use another one would be easier to understand when compared with using `ReentrantLock.Condition`.